ecommerce/extensions/payment/views/hyperpay.py

```diff
@@ -82,36 +135,126 @@ def _verify_status(self, resource_path):
         response_data = response.json()
 
         result_code = response_data['result']['code']
-        if not response.ok or not self.SUCCESSFUL_TRANSACTION_RESULT_CODES_REGEX.search(result_code):
-            logger.error('Received a non-success result code from HyperPay %s', result_code)
-            succeeded = False
+        if not response.ok:
+            logger.error('Received a non-success response status code from HyperPay %s', response.status_code)
+            status = PaymentStatus.FAILURE
+        elif self.PENDING_CHANGEABLE_SOON_CODES_REGEX.search(result_code):
+            logger.warning(
+                'Received a pending status code %s from HyperPay for payment id %s.',
+                result_code,
+                response_data['id']
+            )
+            status = PaymentStatus.PENDING
+        elif self.PENDING_NOT_CHANGEABLE_SOON_CODES_REGEX.search(result_code):
+            logger.warning(
+                'Received a pending status code %s from HyperPay for payment id %s. As this can change '
+                'after several days, treating it as a failure.',
+                result_code,
+                response_data['id']
+            )
+            status = PaymentStatus.FAILURE
+        elif self.SUCCESS_CODES_REGEX.search(result_code):
+            logger.info(
+                'Received a success status code %s from HyperPay for payment id %s.',
+                result_code,
+                response_data['id']
+            )
+        elif self.SUCCESS_MANUAL_REVIEW_CODES_REGEX.search(result_code):
+            logger.warning(
+                'Received a success status code %s from HyperPay which requires manual verification for payment id %s.'
+                'Treating it as a successful transaction.',
+                result_code,
+                response_data['id']
+            )
```

Comment:
This should also send an email after a few days (or be easily searchable within Django admin).

Reply:
@giovannicimolin, the answer is the same as [this one](https://github.com/open-craft/ecommerce/pull/23#discussion_r611834305).